projects/telemetry/inc/nmea.h

```diff
@@ -0,0 +1,96 @@
+// because they do not contain useful information
+typedef enum {
+  NMEA_MESSAGE_ID_UNKNOWN = 0,
+  NMEA_MESSAGE_ID_GGA,
+  NMEA_MESSAGE_ID_GLL,
+  NMEA_MESSAGE_ID_GSA,
+  NMEA_MESSAGE_ID_GSV,
+  NMEA_MESSAGE_ID_RMC,
+  NMEA_MESSAGE_ID_VTG,
+  NUM_NMEA_MESSAGE_ID_TYPES
+} NmeaMessageId;
+
+// An enum to represent a position fix's validity (defined by 1, 2, or 6 as described in
+// https://www.linxtechnologies.com/wp/wp-content/uploads/rxm-gps-f4.pdf
+// on page 14)
+
+typedef enum {
+  NMEA_POSITION_FIX_INVALID = 0,
+  NMEA_POSITION_FIX_GPS_SPS = 1,
+  NMEA_POSITION_FIX_DIFFERENTIAL_GPS_SPS = 2,
+  NMEA_POSITION_FIX_DEAD_RECKONING_MODE = 6,
+  NUM_NMEA_POSITION_FIXES = 4
+} NmeaPositionFix;
+
+typedef struct {
+  uint8_t hh;    // Hours
+  uint8_t mm;    // Minutes
+  uint8_t ss;    // Seconds
+  uint16_t sss;  // Milliseconds
+} NmeaUtcTime;
+
+// Representation of longitude or latitude
+// https://en.wikipedia.org/wiki/Longitude
+typedef struct {
+  int16_t degrees;
+  int16_t minutes;
+  uint16_t fraction;
+} NmeaCoord;
+
+// Info passed from the GPS chip should be dropped into this struct (more fields
+// coming soon)
+
+typedef struct {
+  NmeaPositionFix position_fix;      // True if this struct has valid data
+  int16_t adc;                       // Age of diff. corr. in seconds
+  int16_t drs;                       // Diff. Ref. Station. Not sure what it is yet
+  int16_t hdop_integer;              // Horizontal dilution of precision, characteristic
+  int16_t msl_altitude_integer;      // In meters, characteristic
+  int16_t geoid_seperation_integer;  // In meters, characteristic
+
+  // All fractions in thousandths
+  uint16_t hdop_fraction;              // Horizontal dilution of precision, mantissa
+  uint16_t msl_altitude_fraction;      // In meters, mantissa
+  uint16_t geoid_seperation_fraction;  // In meters, mantissa
+  uint16_t satellites_used;
+
+  NmeaCoord latitude;
+  NmeaCoord longitude;
+  NmeaUtcTime time;
+  char east_west;               // East or West. E for East, W for West
+  char north_south;             // North or South. N for North, S for South
+  char units_msl_altitude;      // Indicated units of above, should be M for meters.
+  char units_geoid_seperation;  // Indicates units of above, should be M for meters.
+  NmeaMessageId message_id;
+} NmeaGgaSentence;
+
+typedef struct {
+  int16_t measure_heading_degrees_integer;    // Whole number of degrees representing the heading
+  uint16_t measure_heading_degrees_fraction;  // Decimal part of degrees representing the heading
+  int16_t speed_kmh_integer;                  // Speed in km/h
+  uint16_t speed_kmh_fraction;                // Speed in km/h, fractional part (after the decimal)
+} NmeaVtgSentence;
+
+// Parsing functions for NMEA sentences
+
+// Tests if the given string is a valid NMEA message. Must be null terminated, will return
+// STATUS_CODE_OK if the message is valid.
+StatusCode nmea_valid(const char *to_check);
```

Comment:
It's likely that the GPS driver will only check message types and validity, but otherwise provide raw output to another telemetry module which would then parse the raw data into something that we can send to the chase car over the xbee / other means when we implement that.

The reason for parsing the messages on the telemetry board is that the telemetry server is structured around CAN messages. It basically only understands data in the form of CAN messages. So basically we're going to send the location data by sending a loopback message or something like that. This will ensure that the telemetry server needs minimal changes to understand the GPS data.